src/status_im/contexts/onboarding/events.cljs

```diff
@@ -1,31 +1,192 @@
 (ns status-im.contexts.onboarding.events
   (:require
     [quo.foundations.colors :as colors]
+    [re-frame.interceptor :as interceptor]
+    [react-native.mmkv :as mmkv]
     status-im.common.biometric.events
     [status-im.constants :as constants]
     [status-im.contexts.shell.constants :as shell.constants]
+    [status-im.feature-flags :as ff]
     [taoensso.timbre :as log]
     [utils.i18n :as i18n]
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
+(defn notifications-setup-start
+  [{:keys [db]}
+   [{:keys [enable-notifications?
+            enable-news-notifications?
+            biometrics?
+            onboarding?
+            syncing?]}]]
+  {:db (if-not onboarding?
+         db
+         (update-in db
+                    [:onboarding/profile]
+                    assoc
+                    :enable-notifications?      enable-notifications?
+                    :enable-news-notifications? enable-news-notifications?))
+   :fx (cond-> []
+         (and (not onboarding?) enable-notifications? enable-news-notifications?)
+         (conj [:dispatch [:notifications/news-notifications-switch true]])
+
+         (and (not onboarding?) enable-notifications?)
+         (conj [:dispatch [:push-notifications/switch true]])
+
+         :always
+         (conj [:dispatch
+                [:onboarding/notifications-setup-done
+                 {:onboarding? onboarding?
+                  :syncing?    syncing?
+                  :biometrics? biometrics?}]]))})
+
+(rf/reg-event-fx :onboarding/notifications-setup-start notifications-setup-start)
+
+(defn notifications-setup-done
+  [{:keys [db]} [{:keys [biometrics? onboarding? syncing?]}]]
+  {:fx (cond-> []
+         (and biometrics? onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finalize-setup]])
+
+         (and biometrics? onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) onboarding? syncing?)
+         (conj [:dispatch [:onboarding/finish-onboarding]])
+
+         (and (not biometrics?) onboarding? (not syncing?))
+         (conj [:dispatch [:onboarding/create-account-and-login]])
+
+         (and (not biometrics?) (not onboarding?) (not syncing?))
+         (conj [:dispatch [:shell/show-root-view]]))})
```

Comment:
This logic was extracted from a few places related to the biometrics onboarding screen, onboarding events, and login events. The gist of this code is to correctly dispatch the continuation event based on whether we're onboarding with or without syncing (and with or without biometrics), or if it's a login.